homeassistant/components/cloud/client.py

```diff
@@ -106,20 +107,35 @@ async def logged_in(self) -> None:
         """When user logs in."""
         await self.prefs.async_set_username(self.cloud.username)
 
-        if self.alexa_config.enabled and self.alexa_config.should_report_state:
+        async def enable_alexa(_):
+            """Enable Alexa."""
             try:
                 await self.alexa_config.async_enable_proactive_mode()
+            except aiohttp.ClientError:  # If no internet available yet
+                async_call_later(self._hass, 30, enable_alexa)
             except alexa_errors.NoTokenAvailable:
                 pass
 
-        if self._prefs.google_enabled:
+        async def enable_google(_):
+            """Enable Google."""
             gconf = await self.get_google_config()
 
             gconf.async_enable_local_sdk()
 
             if gconf.should_report_state:
                 gconf.async_enable_report_state()
 
+        tasks = []
+
+        if self.alexa_config.enabled and self.alexa_config.should_report_state:
+            tasks.append(enable_alexa)
+
+        if self._prefs.google_enabled:
+            tasks.append(enable_google)
+
+        for task in tasks:
+            await task(None)
```

Comment:
Shouldn't we use `asyncio.gather` there so we don't do await in for-loops?
Or does it not matter when the max is so low?

Reply:
Yeah probably. I don't think it matters much here, we're probably at 1 task max.

Reply:
Fixed.